airbrakes/data_handling/apogee_predictor.py

```diff
@@ -128,17 +142,32 @@ def stop(self) -> None:
         Stops the prediction process.
         """
         # Waits for the process to finish before stopping it
-        self._prediction_queue.put(STOP_SIGNAL)  # Put the stop signal in the queue
+        self._processor_data_packet_queue.put(STOP_SIGNAL)  # Put the stop signal in the queue
         self._prediction_process.join()
 
-    def update(self, processed_data_packets: list[ProcessedDataPacket]) -> None:
+    def update(self, processor_data_packets: list[ProcessorDataPacket]) -> None:
         """
-        Updates the apogee predictor to include the most recent processed data packets. This method
+        Updates the apogee predictor to include the most recent processor data packets. This method
         should only be called during the coast phase of the rocket's flight.
 
-        :param processed_data_packets: A list of ProcessedDataPacket objects to add to the queue.
+        :param processor_data_packets: A list of ProcessorDataPacket objects to add to the queue.
         """
-        self._prediction_queue.put_many(processed_data_packets)
+        self._processor_data_packet_queue.put_many(processor_data_packets)
+
+    def get_prediction_data_packets(self) -> list[ApogeePredictorDataPacket]:
+        """
+        Gets the apogee prediction data packets from the queue.
+        """
+        total_packets = []
+        # get_many doesn't actually get all of the packets, so we need to keep checking until
+        # there are no more packets left
+        with contextlib.suppress(Empty):
+            while True:
+                new_packets = self._apogee_predictor_packet_queue.get_many(block=False)
+                if not new_packets:
+                    break
+                total_packets.extend(new_packets)
```

Comment:
I don't think we need to do a while loop here, we'll only ever practically get ~<2 packets per iteration